src/daft-connect/src/display.rs

```diff
@@ -0,0 +1,326 @@
+use std::fmt::Write;
+
+use daft_core::prelude::*;
+
+pub fn to_tree_string(schema: &Schema) -> eyre::Result<String> {
+    let mut output = String::new();
+    // Start with root
+    writeln!(&mut output, "root")?;
+    // Now print each top-level field
+    for (name, field) in &schema.fields {
+        print_field(&mut output, name, &field.dtype, /*nullable*/ true, 1)?;
+    }
+    Ok(output)
+}
+
+// A helper function to print a field at a given level of indentation.
+// level=1 means a single " |-- " prefix, level=2 means
+// " |    |-- " and so on, mimicking Spark's indentation style.
+// A helper function to print a field at a given level of indentation.
+fn print_field(
+    w: &mut String,
+    field_name: &str,
+    dtype: &DataType,
+    nullable: bool,
+    level: usize,
+) -> eyre::Result<()> {
+    let indent = if level == 1 {
+        " |-- ".to_string()
+    } else {
+        format!(" |{}-- ", "    |".repeat(level - 1))
+    };
+
+    let dtype_str = type_to_string(dtype);
+    writeln!(
+        w,
+        "{}{}: {} (nullable = {})",
+        indent, field_name, dtype_str, nullable
+    )?;
+
+    if let DataType::Struct(fields) = dtype {
+        for field in fields {
+            print_field(w, &field.name, &field.dtype, true, level + 1)?;
+        }
+    }
+
+    Ok(())
+}
+
+fn type_to_string(dtype: &DataType) -> String {
+    // We want a nice, human-readable type string.
+    // Spark generally prints something like "integer", "string", etc.
+    // We'll follow a similar style here:
+    match dtype {
+        DataType::Null => "null".to_string(),
+        DataType::Boolean => "boolean".to_string(),
+        DataType::Int8
+        | DataType::Int16
+        | DataType::Int32
+        | DataType::Int64
+        | DataType::UInt8
+        | DataType::UInt16
+        | DataType::UInt32
+        | DataType::UInt64 => "integer".to_string(), // Spark doesn't differentiate sizes
+        DataType::Float32 | DataType::Float64 => "double".to_string(), // Spark calls all floats double for printing
+        DataType::Decimal128(_, _) => "decimal".to_string(),
+        DataType::Timestamp(_, _) => "timestamp".to_string(),
+        DataType::Date => "date".to_string(),
+        DataType::Time(_) => "time".to_string(),
+        DataType::Duration(_) => "duration".to_string(),
+        DataType::Interval => "interval".to_string(),
+        DataType::Binary => "binary".to_string(),
+        DataType::FixedSizeBinary(_) => "fixed_size_binary".to_string(),
+        DataType::Utf8 => "string".to_string(),
+        DataType::FixedSizeList(_, _) => "array".to_string(), // Spark calls them arrays
```

Comment:
i would represent this as a custom type similar to the other non native dtypes.

Reply:
if there is no standard thoughts on something like `daft[fixed_size_list]`?

Reply:
since there seems to be no standard, I'd prefer to separate them into 2 categories. 

### arrow native datatypes:

for arrow native datatypes such as unsigned integers,fsl, etc, lets go with `arrow.<datatype>` such as 
- u64 -> `arrow.uint64`, 
- fsl(u8, 1) -> `arrow.fixed_size_list(1)\n --- element: arrow.u8`

_(this resembles how arrow does extension types)_


### custom daft datatypes:

for non arrow native ones such as SparseTensor, Image, and so on, let's prefix them with daft. 

- image -> `daft.image(<image_mode>)` ex: `daft.image(RGB)`
- sparsetensor(u8) -> `daft.sparse_tensor\n --- element: arrow.u8`

Reply:
should be done